test/proposal.test.ts

```diff
@@ -306,3 +320,130 @@ describe("proposal", function () {
+    gnosisSafeManager = await GnosisSafeManager.initDeterministic(deployer);
+    const IAMB = await artifacts.readArtifact("IAMB");
+    arbitraryMessageBridge = await waffle.deployMockContract(
+      deployer,
+      IAMB.abi,
+    );
+    await arbitraryMessageBridge.mock.requireToPassMessage.returns(messageID);
+    gnosisDao = await (
+      await gnosisSafeManager.newSafe([gnosisDaoOwner.address], 1)
+    ).connect(executor);
+  });
+  const cowDaoSettings: SafeCreationSettings = {
+    owners: [1, 2, 3, 4, 5].map((i) => "0x".padEnd(42, i.toString())),
+    threshold: 5,
+  };
+  const teamConrollerSettings: SafeCreationSettings = {
+    owners: [6, 7, 8].map((i) => "0x".padEnd(42, i.toString())),
+    threshold: 2,
+  };
+  const virtualTokenCreationSettings: VirtualTokenCreationSettings = {
+    merkleRoot: "0x" + "42".repeat(32),
+    usdcToken: "0x0000" + "42".repeat(17) + "01",
+    gnoToken: "0x0000" + "42".repeat(17) + "02",
+    gnoPrice: "31337",
+    wrappedNativeToken: "0x0000" + "42".repeat(17) + "03",
+    nativeTokenPrice: "42424242",
+  };
+
+  let settings: DeploymentProposalSettings;
+  describe("relay of safe deployment", function () {
+    let gnosisSafeDefaults: SafeDeploymentAddresses;
+    before(async function () {
+      const bridgeParameters: BridgeParameter = {
+        multiTokenMediatorGnosisChain: "0x" + "01".repeat(20),
+        multiTokenMediatorETH: "0x" + "02".repeat(20),
+        arbitraryMessageBridgeETH: arbitraryMessageBridge.address,
+      };
+      settings = {
+        gnosisDao: gnosisDao.address,
+        cowDao: cowDaoSettings,
+        teamController: teamConrollerSettings,
+        cowToken: {},
+        virtualCowToken: virtualTokenCreationSettings,
+        bridge: bridgeParameters,
+      };
+      gnosisSafeDefaults = gnosisSafeManager.getDeploymentAddresses();
+    });
+    const expectedCowDaoAddress = "0x6a54ef9C6BE1aF4099336C3bDBBDf690d0B67A7c";
+    it("has the correct target address", async function () {
+      const bridgedGnosisSafeDeployment = await createTxForBridgedSafeSetup(
+        expectedCowDaoAddress,
+        {
+          arbitraryMessageBridgeETH: settings.bridge.arbitraryMessageBridgeETH,
+        },
+        settings.cowDao,
+        gnosisSafeDefaults,
+        hre.ethers,
+      );
+      expect(bridgedGnosisSafeDeployment.to).to.be.equal(
+        settings.bridge.arbitraryMessageBridgeETH,
+      );
+    });
+    it("fails if address is not correct", async function () {
+      await expect(
+        createTxForBridgedSafeSetup(
+          "0x" + "42".repeat(20),
+          {
+            arbitraryMessageBridgeETH:
+              settings.bridge.arbitraryMessageBridgeETH,
+          },
+          settings.cowDao,
+          gnosisSafeDefaults,
+          hre.ethers,
+        ),
+      ).to.be.rejectedWith(Error);
+    });
+    it("has the txData that allows to deploy the safe with correct threshold and owners", async function () {
```

Comment:
I don't understand how the numbers were computed, but otherwise it's a good test!